search-relevancy-test/src/search_relevancy_test/result_logger.clj

```diff
@@ -0,0 +1,115 @@
+(ns search-relevancy-test.result-logger
+  "Functions for writing test results to file"
+  (:require
+   [clj-time.core :as time]
+   [clj-time.format :as time-format]
+   [clojure.data.csv :as csv]
+   [clojure.java.io :as io]
+   [clojure.string :as string]
+   [search-relevancy-test.core :as core]
+   [search-relevancy-test.reporter :as reporter]))
+
+(def local-test-run-csv
+  "local_test_runs.csv")
+
+(def test-run-history-csv
+  "test_run_history.csv")
+
+(defn- get-last-run
+  "Read the CSV file and get the data from the last run, if exists"
+  [csv-file]
+  (when-let [last-run (->> (core/read-csv local-test-run-csv)
+                           (map #(assoc % :date (time-format/parse (:date %))))
+                           (group-by :date)
+                           sort
+                           last)]
+    (val last-run)))
+
+(defn- result-summary->csv-row
+  "Create the CSV row data from the result summary"
+  [result-summary]
+  {:anomaly "SUMMARY"
+   :test ""
+   :description ""
+   :pass (:pass result-summary)
+   :discounted-cumulative-gain (:average-dcg result-summary)
+   :reciprocal-rank (:mean-reciprocal-rank result-summary)
+   :result-description (:result-description result-summary)})
+
+(defn- compare-run
+  "Compare the result to the given run. If they have the same anomaly # and
+  test #, they match. Return true if they match and the discounted-cumulative-gain
+  and reciprocal-rank are different"
+  [result run]
+  (and (= (:anomaly run) (:anomaly result)
+          (= (:test run) (:test result))
+          (or (not= (:discounted-cumulative-gain run)
+                    (:discounted-cumulative-gain result))
+              (not= (:reciprocal-rank run)
+                    (:reciprocal-rank result))))))
```

Comment:
Could you line up all the "and" conditions?  Also, I think you missed a right parenthesis:
(and (= (:anomaly run) (:anomaly result) missing a right parenthesis.